### PR TITLE
Correction for late binding to local listener on api server

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
Amends #1223

### CODE REVIEW

File: libapicore/ApiServer.cpp [hunk 1]
```diff
@@ -17,17 +17,37 @@ void ApiServer::start()
...
+	// Try to bind to port number
+	// if exception occurs it may be due to the fact that
+	// requested port is already in use by another service
+	try
+	{
+		m_acceptor.open(endpoint.protocol());
+		m_acceptor.bind(endpoint);
+		m_acceptor.listen(64);
+	}
+	catch (const std::exception& _e)
```
Comment (XhmikosR) [inline libapicore/ApiServer.cpp]: @AndreaLanfranchi: `_e` is unreferenced here; MSVC complains about it :)

Comment (AndreaLanfranchi) [reply libapicore/ApiServer.cpp]: It's a very strange complaint as it's quite common standard to catch exception by reference. Do you mean it wants it by value ?

Comment (AndreaLanfranchi) [reply libapicore/ApiServer.cpp]: http://www.cplusplus.com/doc/tutorial/exceptions/

Comment (XhmikosR) [reply libapicore/ApiServer.cpp]: See the CI https://ci.appveyor.com/project/ethereum-mining/ethminer/build/job/9573u5xgtp57xatv#L974

Comment (XhmikosR) [reply libapicore/ApiServer.cpp]: Maybe just using one of MSVC's macros like `UNREFERENCED_PARAMETER(_e);` should do the trick.

Comment (AndreaLanfranchi) [reply libapicore/ApiServer.cpp]: Will the directive be accepted by other compilers ? Or should I enclose it in a conditional block if _WIN32 ?

Comment (XhmikosR) [reply libapicore/ApiServer.cpp]: This macro is MSVC specific, so we probably need to wrap it.

Comment (chfast) [reply libapicore/ApiServer.cpp]: Just omit the name `_e`.

Comment (jean-m-cyr) [reply libapicore/ApiServer.cpp]: @chfast approach is commonly used.

Or, you can just add a "(void)_ec;" statement in the block. It doesn't generate any machine code and will satisfy the compiler.

**Pull request merged**